libraries/custom_appbase/include/eosio/chain/exec_pri_queue.hpp

```diff
@@ -68,6 +63,19 @@ class exec_pri_queue : public boost::asio::execution_context
 
 public:
 
+   // only call when no lock required
+   bool execute_highest()
+   {
+      if( !handlers_.empty() ) {
+         auto t = pop();
+         bool empty = handlers_.empty();
+         t->execute();
+         return !empty;
+      }
+
+      return false;
+   }
```

Comment:
No need to change, but just to chat I would have written it like that.

```suggestion
   bool execute_highest()
   {
      bool empty = handlers_.empty();
      if( !empty ) {
         auto t = pop();
         empty = handlers_.empty();
         t->execute();
      }

      return !empty;
   }
```